tasks.py

```diff
@@ -153,60 +255,113 @@ def post_upgrade(context):
     - clearsessions
     - invalidate all
     """
-    docker_compose(context, "run nautobot nautobot-server post_upgrade")
+    command = "nautobot-server post_upgrade"
+
+    if local is None:
+        local = context.nautobot.local
+    if is_truthy(local):
+        context.run(command)
+    docker_compose(context, f"run nautobot {command}")
 
 
 # ------------------------------------------------------------------------------
 # TESTS
 # ------------------------------------------------------------------------------
 @task(
     help={
-        "autoformat": "Apply formatting recommendations automatically, rather than failing if formatting is incorrect."
+        "autoformat": "Apply formatting recommendations automatically, rather than failing if formatting is incorrect.",
+        "local": "run this task locally vs inside the docker container (default: False)",
     }
 )
-def black(context, autoformat=False):
+def black(context, autoformat=False, local=None):
     """Check Python code style with Black."""
+    if local is None:
+        local = context.nautobot.local
     if autoformat:
         black_command = "black"
     else:
         black_command = "black --check --diff"
-    docker_compose(
-        context,
-        f"run --entrypoint '{black_command} development/ nautobot/ tasks.py' nautobot",
-        pty=True,
-    )
 
+    command = f"{black_command} development/ nautobot/ tasks.py"
 
-@task
-def flake8(context):
+    if is_truthy(local):
+        context.run(command)
+    else:
+        docker_compose(
+            context,
+            f"run --entrypoint '{command}' nautobot",
+            pty=True,
+        )
+
+
+@task(help={"local": "run this task locally vs inside the docker container (default: False)"})
+def flake8(context, local=None):
     """Check for PEP8 compliance and other style issues."""
-    docker_compose(context, "run --entrypoint 'flake8 development/ nautobot/ tasks.py' nautobot", pty=True)
+    command = "flake8 development/ nautobot/ tasks.py"
+
+    if local is None:
+        local = context.nautobot.local
+    if is_truthy(local):
+        context.run(command)
+    else:
+        docker_compose(context, f"run --entrypoint '{command}' nautobot", pty=True)
+
+
+@task(help={"local": "run this task locally vs inside the docker container (default: False)"})
+def check_migrations(context, local=None):
+    """Check for missing migrations."""
+    command = "nautobot-server --config=nautobot/core/tests/nautobot_config.py makemigrations --dry-run --check"
```

Comment:
Why are we using the test `nautobot_config.py` for this task? We don't use it for `makemigrations` or `migrate`.

Reply:
I believe this is used strictly by `cibuild.sh` this task was there and simply copied over.  I'm fine if we want to remove it but I think we should also consider removing it from [unittest](https://github.com/nautobot/nautobot/pull/211/files#diff-2e746c9eacad6e2c2fdefdc6a665b5ce4607ea384ebf97a729044b79acc778d6R331-R334) as well

Reply:
Correct, only for test as done inside of `scripts/cibuild.sh` currently.